src/sensai/featuregen.py

```diff
@@ -24,28 +24,45 @@ class FeatureGenerator(ABC):
     from an input DataFrame
     """
     def __init__(self, categoricalFeatureNames: Sequence[str] = (),
-            normalisationRules: Sequence[data_transformation.DFTNormalisation.Rule] = (), addCategoricalDefaultRules=True):
+                 normalisationRules: Sequence[data_transformation.DFTNormalisation.Rule] = (),
+                 generatedColumnsRuleTemplate: data_transformation.DFTNormalisation.RuleTemplate = None, addCategoricalDefaultRules=True):
         """
         :param categoricalFeatureNames: if provided, will ensure that the respective columns in the generated data frames will
             have dtype 'category'.
             Furthermore, presence of meta-information can later be leveraged for further transformations, e.g. one-hot encoding.
         :param normalisationRules: Rules to be used by DFTNormalisation (e.g. for constructing an input transformer for a model).
             These rules are only relevant if a DFTNormalisation object consuming them is instantiated and used
             within a data processing pipeline. They do not affect feature generation.
+        :param generatedColumnsRuleTemplate: If given, a normalisation rule based on this template will be created for all generated columns.
+            As with other rules, this is only relevant if DFTNormalization is used after feature generation.
+            The rule template should only be provided if no categoricalFeatureNames and normalisationRules have been passed
         :param addCategoricalDefaultRules:
             If True, normalisation rules for categorical features (which are unsupported by normalisation) and their corresponding one-hot
             encoded features (with "_<index>" appended) will be added.
         """
+        self.generatedColumnNames = None
```

Comment:
We should make clear that it's the list of column names that was generated by the most recent generation that this object performed. To make this clear we at least need to rename this column; even more clarity can be achieved by using a getter instead and making this attribute private, such that we can add the necessary documentation.

Reply:
Done